src/audio/rtnr/rtnr.c

```diff
@@ -0,0 +1,554 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Realtek Semiconductor Corp. All rights reserved.
+//
+// Author: Ming Jen Tai <mingjen_tai@realtek.com>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/rtnr/rtnr.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define RTNR_ENABLE_PROCESSING  1
+
+#if RTNR_ENABLE_PROCESSING
+#include <sof/audio/rtnr/rtklib/include/Ringbuffer_new.h>
+#include <sof/audio/rtnr/rtklib/include/RTK_MA_API.h>
+#endif // RTNR_ENABLE_PROCESSING
+
+#if RTNR_ENABLE_PROCESSING
+
+#define SAMPLING_RATE		16000
+#define PROCESS_RATE		16000
+#define PRESET_SIZE			1040
+#define MicNum				2
+#define SpkNum				2
+
+#if SAMPLING_RATE == 48000
```

Comment:
sample rate is a runtime construct, you should be handling this at runtime or rejecting parameters you don't support